### PR TITLE
Use localstorage only to initialize settings map

### PR DESCRIPTION
**Pull request by andrwwbstr**
This makes query overrides temporary (i.e. they do not change the value, if any, in local storage).  I believe it also solves #799.

### CODE REVIEW

File: app/ui.js [hunk 1]
```diff
@@ -734,7 +734,7 @@ var UI = {
 
         // Save the cookie for this session
         if (typeof value !== 'undefined') {
-            WebUtil.writeSetting(name, value);
+            WebUtil.writeSetting(name, value, true);
```
Comment (CendioOssman) [inline app/ui.js]: The only place that provides a `value` is the call from `initSetting()`. So I think we can take this opportunity to move this chunk to `initSetting()` and get rid of the `value` argument.

File: app/webutil.js [hunk 2]
```diff
@@ -121,22 +121,22 @@ export function initSettings (callback /*, ...callbackArgs */) {
             }
         });
     } else {
-        // No-op
+        settings = {};
```
Comment (CendioOssman) [inline app/webutil.js]: Why is this added?

File: app/webutil.js [hunk 3]
```diff
@@ -121,22 +121,22 @@ export function initSettings (callback /*, ...callbackArgs */) {
...
         if (callback) {
             callback.apply(this, callbackArgs);
         }
     }
 };
 
 // No days means only for this browser session
-export function writeSetting (name, value) {
+export function writeSetting (name, value, temporary) {
```
Comment (CendioOssman) [inline app/webutil.js]: "write" and "temporary" seems like a confusing mix to me. Perhaps introduce a new `setSetting()` that only updates the `settings` object?

File: app/webutil.js [hunk 4]
```diff
@@ -162,9 +163,9 @@ export function readSetting (name, defaultValue) {
 
 export function eraseSetting (name) {
     "use strict";
+    settings[name] = undefined;
```
Comment (CendioOssman) [inline app/webutil.js]: Why not `delete` to get things fully back to before it was ever set?

File: app/webutil.js (hunk 2 above)
Comment (andrwwbstr) [reply app/webutil.js]: Mostly for the tests so that the settings object can be reset between tests.  However, if `initSettings` were to be called again (which it isn't), it makes sense to reset the settings object.

File: app/webutil.js (hunk 4 above)
Comment (andrwwbstr) [reply app/webutil.js]: If the setting was deleted, then the next time it is read it would be restored back to the value in localstorage.  If localstorage was modified by another tab between the erase and the next read, then the read value would be inconsistent.

It may be a moot point though since it seems that `eraseSetting` is never used.  Would it just be better to remove it instead?

Comment (CendioOssman) [reply app/webutil.js]: Not entirely clear what we want here since it isn't called. But if I read the code correctly this will nuke it from local storage as well. So the possibly confusing case is the one you describe, when another tab does a write after the first tab has done an erase.

I think a `delete` would be safer for now and we can look over this once the API starts getting use. You could add a comment about this confusion there if you'd like.

File: app/webutil.js (hunk 2 above)
Comment (CendioOssman) [reply app/webutil.js]: Ah, makes sense.

File: app/webutil.js [hunk 5]
```diff
@@ -128,12 +128,17 @@ export function initSettings (callback /*, ...callbackArgs */) {
     }
 };
 
+// Update the settings cache, but do not write to permanent storage
+export function setSetting (name, value) {
+    "use strict";
```
Comment (CendioOssman) [inline app/webutil.js]: These are no longer needed, so let's not add new ones.

**Comment by CendioOssman** [comment]
Well, rather it will be okay to merge once 1.0.0 is out.

**Comment by CendioOssman** [comment]
Do you think you could rebase those into the original commits? :)
